UnityProject/Assets/Scripts/Systems/Explosions/NodeTypes/ExplosionEmpNode.cs

```diff
@@ -90,6 +90,11 @@ private bool IsEmpAble(GameObject thing)
 				if (machine.isEMPResistant) return false;
 			}
 
+			if (thing.TryGetComponent<BodyPartFunctionality>(out var bpf))
+			{
+				if(bpf.isEMPVunerable == false) return false;
+			}
```

Comment:
Could this check be inside `BodyPartFunctionality` instead? As far as I can tell, it just calls `OnEmp` if this returns true.